torch/csrc/jit/script/edit_distance.h

```diff
@@ -0,0 +1,50 @@
+#include <algorithm>
+#include <memory>
+
+namespace torch { namespace jit { namespace script {
+
+// computes levenshtein edit distance between two words
+// returns maxEditDistance + 1 if the edit distance exceeds MaxEditDistance
+// reference: http://llvm.org/doxygen/edit__distance_8h_source.html
+size_t computeEditDistance(const std::string& word1, const std::string& word2,
+    size_t maxEditDistance) {
+
+  size_t m = word1.size();
+  size_t n = word2.size();
+
+  const unsigned small_buffer_size = 64;
+  unsigned small_buffer[small_buffer_size];
+  std::unique_ptr<unsigned[]> allocated;
```

Comment:
We could... I think part of the reason of using an existing implementation is to avoid the cognitive overhead of rethinking & retesting it. The more you change the implementation the less that applies

Reply:
I would argue that you *should* be rethinking and retesting code that you are copying from elsewhere.